rama-http/src/io/upgrade.rs

```diff
@@ -126,6 +132,7 @@ impl Upgraded {
     {
         Self {
             io: Rewind::new_buffered(Box::new(io), read_buf),
+            extensions: Extensions::new(),
```

Comment:
In future PR this will need the extensions from IO as its parent in a frozen state. Similar to how `BytesMut` has `freeze` we probably also need something like that for Extensions, which should basically clone `self` and return a new extensions with self as a arc or something such that the fallback to parents also work (as deep as possible).

Just leaving it as a note, you can resolve this as-is once you read it.

Reply:
Not really. Shouldn't make a difference for h2. What happens with h2 is that just that stream is tunneling. That stream is what I called pseudo connections. In case of h2 your hierarchy is

```
tcp
   |
  stream
     |
     http req 
```

So your IO Extensions will in that case be `stream`, with the tcp one already frozen and shared among all streams. That's something that `http` core should be able to facilitate.